### DISCUSSION TITLE
How to set minimum garbage collection for kubelet on k3s

### DISCUSSION THREAD

**Discussion by SarthakNarayan** [k3s-io/k3s#8525] [Q&A] [answered]
Hi,

I know I can set kubelet parameters on k3s using `--kubelet-arg`. I came to know that the `image-gc-high-threshold` and `image-gc-low-threshold` have been deprecated. I want my k3s cluster to remove unused images if the disk space is less than 70%. Any idea how I can set it?
The following [link](https://kubernetes.io/docs/concepts/architecture/garbage-collection/) suggests that I should use `HighThresholdPercent` and `LowThresholdPercent` to control image garbage collection thresholds. Can I pass them to k3s as `--kubelet-arg` or have I misunderstood something?

Thanks

**Comment by brandond** [comment]
Yes, just use kubelet-arg. Passing these args to the kubelet may be deprecated but shows no signs of actually going away.

**Comment by SarthakNarayan** [comment]
I am using the following command to set up my k3s cluster. But it is not cleaning up the unused images. For testing I just run 2 pods and then delete one of them but I can still see the image of the deleted pod when I do `sudo crictl images`.

```bash
curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.27.3+k3s1 sh -s - \
  --write-kubeconfig ~/.kube/config \
  --write-kubeconfig-mode 600 \
  --kubelet-arg "--image-gc-high-threshold=10" \
  --kubelet-arg "--image-gc-low-threshold=5" 
```

**Reply by brandond** [reply, marked as answer]
Yeah, its periodic, not immediate... those are the correct args but I think you might have unrealistic expectations around how aggressively images are cleaned up to reach the target disk utilization.

**Reply by SarthakNarayan** [reply]
I see. Thanks a lot for answering my queries.
